RecoLocalMuon/GEMSegment/plugins/ME0SegAlgoRU.cc

```diff
@@ -22,447 +22,536 @@
+                                          HitAndPositionContainer::const_iterator i2) const {
+  // Iterate over the layers with hits in the chamber
+  // Skip the layers containing the segment endpoints
+  // Test each hit on the other layers to see if it is near the segment
+  // If it is, see whether there is already a hit on the segment from the same layer
+  //    - if so, and there are more than 2 hits on the segment, copy the segment,
+  //      replace the old hit with the new hit. If the new segment chi2 is better
+  //      then replace the original segment with the new one (by swap)
+  //    - if not, copy the segment, add the hit. If the new chi2/dof is still satisfactory
+  //      then replace the original segment with the new one (by swap)
+
+  //Hits are ordered by layer, "i1" is the inner hit and i2 is the outer hit
+  //so possible hits to add must be between these two iterators
+  for (auto iH = i1 + 1; iH != i2; ++iH) {
+    if (iH->layer == i1->layer)
+      continue;
+    if (iH->layer == i2->layer)
+      break;
+    if (used[iH->idx])
+      continue;
+    if (!areHitsConsistentInTime(maxTOF, proto_segment, *iH))
+      continue;
+    if (!isHitNearSegment(maxETA, maxPhi, current_fit, proto_segment, *iH))
+      continue;
+    if (hasHitOnLayer(proto_segment, iH->layer))
+      compareProtoSegment(current_fit, proto_segment, *iH);
+    else
+      increaseProtoSegment(maxChi2, current_fit, proto_segment, *iH);
+  }
 }
 
-bool ME0SegAlgoRU::areHitsCloseInEta(const float maxETA, const bool beamConst, const GlobalPoint& h1, const GlobalPoint& h2) const {
-	float diff = 0;
-	diff = std::fabs(h1.eta() - h1.eta());
-	edm::LogVerbatim("ME0SegAlgoRU") << "[ME0SegAlgoRU::areHitsCloseInEta] gp1 = "<<h1<<" in eta part = "<<h1.eta() <<" and gp2 = "<<h2<<" in eta part = "<<h2.eta() <<" ==> dEta = "<<diff<<" ==> return "<<(diff < 0.1)<<std::endl;
-	return (diff < std::max(maxETA,float(0.01)) ); //temp for floating point comparision...maxEta is the difference between partitions, so x1.5 to take into account non-circle geom.
+bool ME0SegAlgoRU::areHitsCloseInEta(const float maxETA,
+                                     const bool beamConst,
+                                     const GlobalPoint& h1,
+                                     const GlobalPoint& h2) const {
+  float diff = 0;
+  diff = std::fabs(h1.eta() - h1.eta());
+  edm::LogVerbatim("ME0SegAlgoRU") << "[ME0SegAlgoRU::areHitsCloseInEta] gp1 = " << h1 << " in eta part = " << h1.eta()
+                                   << " and gp2 = " << h2 << " in eta part = " << h2.eta() << " ==> dEta = " << diff
+                                   << " ==> return " << (diff < 0.1) << std::endl;
+  return (
+      diff <
+      std::max(
+          maxETA,
+          float(
+              0.01)));  //temp for floating point comparision...maxEta is the difference between partitions, so x1.5 to take into account non-circle geom.

```

Comment:
@smuzaffar @fabiocos 
The new version looks quite ridiculous.
I guess things like these will have to be redone manually.

Reply:
I wonder if there would be an automated way to move the comment to preceding line in cases like this (assuming it is the comment that causes the newlines).

Reply:
yes it is due to the comment. clang-format tries to keep the line length to under 120 chars. We have seen this type of changes and yes one has to manually fix it e.g. move the comment above the line and keep the original code under 120 chars.

Reply:
@smuzaffar : could it get fixed manually now, then? (This one and all other similar cases)
I'm pretty sure that if we merge the PR as such, then most of these "ridiculous" (and barely readable) lines will never get fixed.

Reply:
we are probably past the middle of the campaign and perhaps similar cases were already rubber-stamped in other PRs.
It may be more effective to update the rules and then run an update on the full code-base.

Reply:
@slava77 @perrotta we had such a case in some L1/DQM code, and it was manually adjusted. Of course adding a check, if possible, would be good for the future, but here I would say we may just fix it "by hand"

Reply:
for another case you may just see #26840 ...